costar_hyper/costar_hypertree_inference.py

```diff
@@ -0,0 +1,185 @@
+        '''
+        Generates plots for the given metric
+        #Arguments
+        score_file: name of the file containing the metrics
+        metric_2: name of metric to be used in plot generation
+        '''
+        with open(score_file, 'r') as fp:
+            reader = csv.reader(fp)
+            headers = next(reader, None)
+            scores = list(reader)
+        # metric_1_index = headers.index(metric_1)
+        metric_2_index = headers.index(metric_2)
+        frames = []
+        loss = []
+        for row in scores:
+            frames.append(row[1])
+            loss.append(row[metric_2_index])
+
+        print(headers)
+        frames = list(map(int, frames))
+        loss = list(map(float, loss))
+        figure1 = plt.figure(1, figsize=(20, 10))
+        plt.xticks(np.arange(min(frames), max(frames)+1, 10))
+        plt.yticks(np.arange(min(loss), max(loss)+1, 0.1))
+        indexes = np.where(np.array(frames) == 1)[0]
+        # print(indexes)
+        ax = plt.axes()
+        n_lines = len(indexes)
+        ax.set_color_cycle([plt.cm.cool(i) for i in np.linspace(0, 1, n_lines)])
+        count = 0
+        for i in indexes[1:]:
+            goals = self.gripper_action_goal_idx[count]
+            count += 1
+            # print(len(goals))
+            # print(len(frames[indexes[count-1]:i]))
+            plt.scatter(np.array(goals[1:] - np.array(frames[indexes[count-1]:i])), loss[indexes[count-1]:i])
+        goals = self.gripper_action_goal_idx[-1]
+        # print(goals)
+        plt.scatter(np.array(goals[1:]) - frames[indexes[-1]:], loss[indexes[-1]:])
+        # print(frames[indexes[-1]:-8]-np.array(goals[1:]))
+        # plt.plot(frames[:225], loss[:225])
+        # plt.plot(frames[225:], loss[225:])
+        plt.xlabel('Distance to goal')
+        plt.ylabel(metric_2)
+        plt.savefig("plot1.png")
+        plt.show()
+        # plt.show()
+        figure2 = plt.figure(2, figsize=(20, 10))
+        frame_range = range(1, len(frames)+1)
+        print(len(frame_range))
+        plt.xticks(np.arange(min(loss), max(loss)+1, 0.1))
+        figure2.axes[0].yaxis.set_major_formatter(ticker.PercentFormatter(xmax=len(frames)))
+        plt.plot(np.sort(loss), frame_range)
+        # print(figure2.axes)
+        plt.xlabel(metric_2)
+        plt.ylabel('Examples')
+        plt.savefig('plot2.png')
+        plt.show()
+
+
+if __name__ == "__main__":
+
+    filenames = glob.glob(os.path.expanduser(r'C:\Users\Varun\JHU\LAB\Projects\costar_task_planning_stacking_dataset_v0.1/*success.h5f'))
+    output_shape = (224, 224, 3)
+    load_weights = "2018-09-04-20-17-25_train_v0.3_msle-vgg_semantic_rotation_regression_model--dataset_costar_block_stacking-grasp_goal_aaxyz_nsc_5-epoch-412-val_loss-0.002-val_angle_error-0.279.h5"
+    # filenames_updated, file_len_list = inference_mode_gen(filenames[:2])
+    print(len(filenames))
+    training_generator = CostarBlockStackingSequence(
```

Comment:
were there any changes to the block stacking sequence class itself?

Reply:
I did make some changes but then changed this code to work without those changes.